src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -658,6 +676,18 @@ def _list_workflow_runs(
                 pageSize=page_size,
                 pageToken=page_token,
                 workspaceId=workspace,
+                maxAge=(int(max_age.total_seconds()) if max_age else None),
+                state=(
+                    ",".join(
+                        (
+                            [state.value]
+                            if not isinstance(state, list)
+                            else [st.value for st in state]
+                        )
+                    )
+                    if state
+                    else None
+                ),
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏This seems just enough complexity to pull out.

```
def _get_state_query(state: Optional[Union[State, List[State]]]):
    if state is None:
        return None
    if isinstance(state, State):
        return state.value
    else:
        return ",".join(st.value for st in state)
```

then call with:
```suggestion
                state=_get_state_query(state),
```

Reply:
[5e87105](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/335/commits/5e871059eb337be15fae6ef2e0055197a2a651ef)